src/reanimated2/shareables.ts

```diff
@@ -237,18 +246,22 @@ export function makeShareableCloneOnUIRecursive<T>(value: T): ShareableRef<T> {
     return value;
   }
   function cloneRecursive<T>(value: T): ShareableRef<T> {
-    const type = typeof value;
-    if ((type === 'object' || type === 'function') && value !== null) {
-      let toAdapt: any;
+    if (
+      (typeof value === 'object' && value !== null) ||
+      typeof value === 'function'
```

Comment:
Let's restore original logic here:
```suggestion
    const type = typeof value;
    if ((type === 'object' || type === 'function') && value !== null) {
```